docs/madr/decisions/014-circuit-breaker-and-outlier-detector-policies.md

```diff
@@ -0,0 +1,192 @@
+# Circuit Breaker and Outlier Detector policies compliant with 2.0 model
+
+* Status: accepted
+
+Technical Story: [#4736](https://github.com/kumahq/kuma/issues/4736)
+
+## Context and Problem Statement
+
+[New policy matching MADR](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md)
+introduces a new approach how Kuma applies configuration to proxies. Rolling out strategy implies creating a new set of
+policies that satisfy a new policy matching. Current MADR aims to define a timeout policy that will be compliant with
+2.0 policies model.
+
+Current `CircuitBreaker` policy combines
+envoy's [circuit breaking](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/circuit_breaking)
+and [outlier detection](https://gstwww.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier) capabilities behind the scene.
+Consequences of this approach may introduce the situation when during reading envoy's stats for debugging purposes,
+you may look for stats prefixed with `cluster.<name>.circuit_breakers.` with occurred fault in mind, and according
+to the behaviour of our policy, if the data plane was ejected because of fault it's taking into account, stats which
+contain interesting us data will be prefixed with `cluster.<name>.outlier_detection.` prefix.
+
+## Considered Options
+
+* Create two separate policies: `CircuitBreaker` and `OutlierDetector`
```

Comment:
No Mesh prefix in the end? Also I think @lobkovilya really wanted to stay away from this option

Reply:
Updated with the prefix (forgot about it). I unfortunately don't remember @lobkovilya wanting to stay away from this option, but I'm definitely open for conversation about it

Reply:
It's not like I "really wanted" :) It's a question if we want to stay closer to Envoy terminology or use something that we believe is more widely adopted. I think when users want CircuitBreaker they actually need `envoy.OutlierDetector`. Having `envoy.CircuitBreaker` and `envoy.OutlierDetector` inside one MeshCircuitBreaker policy seems like a good compromise.

Reply:
After some thinking I agree, so I updated the MADR with appropriate changes